src/Merchants.php

```diff
@@ -142,15 +142,36 @@ public static function update_or_create_merchant() {
 			'merchant_name'    => $merchant_name,
 		);
 
-		// The response only contains the merchant id.
-		$response = API\Base::update_or_create_merchant( $args );
+		$cache_key = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '_request_' . md5( wp_json_encode( $args ) );
+		$cache     = get_transient( $cache_key );
+
+		if ( false !== $cache ) {
+			return $cache;
+		}
+
+		try {
+			// The response only contains the merchant id.
+			$response = Base::update_or_create_merchant( $args );
+		} catch ( Throwable $th ) {
+			$delay = Pinterest_For_Woocommerce()::get_data( 'create_merchant_delay' ) ?? 10 * MINUTE_IN_SECONDS;
+
+			set_transient( $cache_key, $response, $delay );
```

Comment:
If there was an Exception then the `$response` is not defined. The only thing that we can store here is `$th->getCode`. We will be able to use it later in the cache check on line 149 for throwing an exception.